lib/pf/authentication.pm

```diff
@@ -243,19 +243,20 @@ sub authenticate {
     my $message;
     foreach my $current_source (@sources) {
         my $result;
+        my $connection;
         $logger->trace("Trying to authenticate '$username' with source '".$current_source->id."'");
         eval {
-            ($result, $message) = $current_source->authenticate($username, $password);
+            ($result, $message, $connection) = $current_source->authenticate($username, $password);
         };
         # First match wins!
         if ($result) {
             $logger->info("Authentication successful for $username in source ".$current_source->id." (".$current_source->type.")");
-            return ($result, $message, $current_source->id);
+            return ($result, $message, $current_source->id, $connection);
         }
     }
 
     $logger->trace("Authentication failed for '$username' for all ".scalar(@sources)." sources");
-    return ($FALSE, $message ? $message : $AUTH_FAIL_MSG);
+    return ($FALSE, $message ? $message : $AUTH_FAIL_MSG, undef);
```

Comment:
Not sure this does what you want, the 3rd return is the source id not the connection but anyway it ends up as undef so just remove the additional undef